x-team/xp-ui
# Create GenericTooltip

**Release Type:** *non-breaking changes*

Fixes https://x-team-internal.atlassian.net/browse/XP-3456

## Description

### Acceptance Criteria

1. Create a generic Tooltip component (with minimal API requirements at the moment)

> Done! The `GenericTooltip` is an interface of https://github.com/wwayne/react-tooltip. I pick this lib because it's very similar to our needs and very flexible. Styling is not really good and current stage have very little support for different cases beyond defaults, but importing this lib leveraged a huge amount of work.

2. Create showcases in Storybook

> Done!

3. Implement the Tooltip in the HeaderBar component, in the "Update your profile" link (see https://xpui-develop.netlify.com/?path=/story/ui-components-headerbar-use-cases--with-links)

> Done!

4. Consider moving the business information to the consumer app (XP-Reg), but be aware of HeaderBar implementation under the scope of  https://x-team-internal.atlassian.net/browse/XP-3310

> During discussions with @WagnerMoreira, we realised that the HeaderBar menu links required an amount of work that's specific and could be scoped into own task, therefore I suggested to raise a new dev task focused on it: https://x-team-internal.atlassian.net/browse/XP-3459, I think we can consider this AC to be addressed there, in the scope of this PR I just used what already has

5. Make sure that the new generic Tooltip is also compatible and can be re-used in the new WelcomeHero UI component that's going to be created in the scope of  https://x-team-internal.atlassian.net/browse/XP-3367

> It's compatible because it's a generic and flexible Tooltip 👍 


## Checklist


## Steps to Test or Reproduce

1. Make sure to run `npm i`
1. Check new `GenericTooltip` component at http://localhost:9001/?path=/story/core-components-generictooltip--basic-usage
1. Check implementation in `HeaderBar` at http://localhost:9001/?path=/story/ui-components-headerbar-use-cases--with-links

## Impacted Areas in Application

New `GenericTooltip` added. Also extended `HeaderBar`.

## Screenshots

![2019-11-13 18 27 17](https://user-images.githubusercontent.com/131859/68806513-cc19b100-0644-11ea-9db4-0541c940ad4e.gif)

![2019-11-13 18 28 57](https://user-images.githubusercontent.com/131859/68806514-cc19b100-0644-11ea-94d0-f46566f6525d.gif)
